R/tm_g_ae_oview.R

```diff
@@ -285,34 +309,16 @@ srv_g_ae_oview <- function(id,
 
     output_q <- reactive({
       ANL <- data[[dataname]]() # nolint
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
+
+      teal::validate_has_data(ANL, min_nrow = 10, msg = sprintf("%s has not enough data", dataname))
+
+      teal::validate_inputs(iv())
+
       validate(need(
-        is.factor(ANL[[input$arm_var]]),
-        "Selected arm variable needs to be a factor."
-      ))
-      validate(need(input$flag_var_anl, "Please select at least one flag."))
-      iv_comp <- shinyvalidate::InputValidator$new()
-      iv_comp$add_rule("arm_trt", shinyvalidate::sv_not_equal(
-        input$arm_ref,
-        message_fmt = "Must not be equal to Control"
+        input$arm_trt %in% unique(ANL[[input$arm_var]]) ||
+          input$arm_ref %in% unique(ANL[[input$arm_var]]),
```

Comment:
`unique` here is an extra call which just slows down the execution. It's not a big deal, but I can assure you that %in% is smart enough to stop execution on the first occurrence in the vector.

```r
x <- sample(letters, size = 10000, replace = TRUE)


microbenchmark::microbenchmark(
  "a" %in% unique(x),
  "a" %in% x
)
```

```suggestion
        input$arm_trt %in% ANL[[input$arm_var]] ||
          input$arm_ref %in% ANL[[input$arm_var]],
```

Reply:
I think that there should be `&&` instead `||`